posthog/temporal/sentry.py

```diff
@@ -0,0 +1,87 @@
+from dataclasses import is_dataclass
+from typing import Any, Optional, Type, Union
+
+from temporalio import activity, workflow
+from temporalio.worker import (
+    ActivityInboundInterceptor,
+    ExecuteActivityInput,
+    ExecuteWorkflowInput,
+    Interceptor,
+    WorkflowInboundInterceptor,
+    WorkflowInterceptorClassInput,
+)
+
+with workflow.unsafe.imports_passed_through():
+    from sentry_sdk import Hub, capture_exception, set_context, set_tag
+
+
+def _set_common_workflow_tags(info: Union[workflow.Info, activity.Info]):
+    set_tag("temporal.workflow.type", info.workflow_type)
+    set_tag("temporal.workflow.id", info.workflow_id)
+
+
+class _SentryActivityInboundInterceptor(ActivityInboundInterceptor):
+    async def execute_activity(self, input: ExecuteActivityInput) -> Any:
+        # https://docs.sentry.io/platforms/python/troubleshooting/#addressing-concurrency-issues
+        with Hub(Hub.current):
+            set_tag("temporal.execution_type", "activity")
+            set_tag("module", input.fn.__module__ + "." + input.fn.__qualname__)
+
+            activity_info = activity.info()
+            _set_common_workflow_tags(activity_info)
+            set_tag("temporal.activity.id", activity_info.activity_id)
+            set_tag("temporal.activity.type", activity_info.activity_type)
+            set_tag("temporal.activity.task_queue", activity_info.task_queue)
+            set_tag("temporal.workflow.namespace", activity_info.workflow_namespace)
+            set_tag("temporal.workflow.run_id", activity_info.workflow_run_id)
+            try:
+                return await super().execute_activity(input)
+            except Exception as e:
+                if len(input.args) == 1 and is_dataclass(input.args[0]):
```

Comment:
From the way their example code was written, I guess `parse_inputs` has already been run by this point? Either way it's written to be safe if they aren't.